src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs

```diff
@@ -60,6 +60,33 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                         var parameterReferenceSyntax = nullCoalescing.Left;
                         editor.ReplaceNode(nullCoalescing, parameterReferenceSyntax.WithAppendedTrailingTrivia(SyntaxFactory.ElasticMarker));
                         break;
+                    case IfStatementSyntax { Else.Statement: BlockSyntax { Statements: var statementsWithinElse } } ifStatementWithElseBlock:
+                        {
+                            var parent = (BlockSyntax)ifStatementWithElseBlock.Parent!;
+                            var statements = parent.Statements;
+                            var indexOfIfStatement = statements.IndexOf(ifStatementWithElseBlock);
+                            using var _1 = ArrayBuilder<StatementSyntax>.GetInstance(capacity: statements.Count + statementsWithinElse.Count - 1, out var newStatements);
+
+                            for (var i = 0; i < indexOfIfStatement; i++)
+                            {
+                                newStatements.Add(statements[i]);
+                            }
+                            foreach (var statementWithinElse in statementsWithinElse)
+                            {
+                                newStatements.Add(statementWithinElse.WithPrependedLeadingTrivia(SyntaxFactory.ElasticMarker));
+                            }
+                            for (var i = indexOfIfStatement + 1; i < statements.Count; i++)
+                            {
+                                newStatements.Add(statements[i]);
+                            }
+
+                            parent.Statements.ReplaceRange(ifStatementWithElseBlock, newStatements);
+                            editor.ReplaceNode(parent, parent.WithStatements(SyntaxFactory.List(newStatements)));
+                            break;
+                        }
```

Comment:
my general preference is once switch clauses get large, they should be broken out into helper methods.

Reply:
(i would extract both 'ifstatement' cases into one case that delegates to a simple helper that then checks the shape of it).

Reply:
Simplifying this. I think this also addresses several comments below.